### ISSUE TITLE
[@typescript-eslint/no-unused-vars] not respecting varsIgnorePattern

### ISSUE THREAD

**Issue by xenoterracide** [typescript-eslint/typescript-eslint#2249]
false positive @typescript-eslint is detecting an excluded `varsIgnorePattern`

```json
{
  "parser": "@typescript-eslint/parser", // Specifies the ESLint parser
  "plugins": ["@typescript-eslint"],
  "extends": [
    "eslint:recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:import/typescript",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended",
    "prettier/@typescript-eslint" // Uses eslint-config-prettier to disable ESLint rules from @typescript-eslint/eslint-plugin that would conflict with prettier
  ],
  "rules": {
    "import/no-default-export": ["error"],
    "max-lines": ["error"],
    "max-lines-per-function": [
      "error",
      { "max": 30, "skipComments": true, "skipBlankLines": true }
    ],
    "import/no-extraneous-dependencies": ["error"],
    "require-await": ["error"],
    "@typescript-eslint/no-unused-vars": ["error", { "varsIgnorePattern": "^_+" }] // also tried `^_`
  },
  "overrides": [
    {
      "files": ["**/*.test.ts"],
      "rules": {
        "max-lines-per-function": ["off"],
        "@typescript-eslint/no-non-null-assertion": ["off"]
      }
    }
  ]
}
```
```
> npx eslint src/order/controller/resolver/test.ts                         # service-graph -> feature/RS2-1064-initiate-pickup-with-pin $ ! ?

/Users/calebcushing/IdeaProjects/service-graph/src/order/controller/resolver/test.ts
  3:13  error  '_' is defined but never used  @typescript-eslint/no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```
here's the class

```ts
export class Foo {
  method(): (str: string) => string {
    return (_): string => {
      return 'foo';
    };
  }
}
```

```
 npx --eslint --version                                              # service-graph -> feature/RS2-1064-initiate-pickup-with-pin $ ! ? RC=1
6.14.4

 npx eslint --version                                                     # service-graph -> feature/RS2-1064-initiate-pickup-with-pin $ ! ?
v6.8.0

> yarn list @typescript-eslint/eslint-plugin @typescript-eslint/parser typescript
yarn list v1.22.4
warning Filtering by arguments is deprecated. Please use the pattern option instead.
├─ @typescript-eslint/eslint-plugin@2.32.0
├─ @typescript-eslint/parser@2.32.0
└─ typescript@3.7.4
✨  Done in 1.69s.

> yarn list typescript @typescript-eslint/eslint-plugin @typescript-eslint/parser eslint
yarn list v1.22.4
warning Filtering by arguments is deprecated. Please use the pattern option instead.
├─ @typescript-eslint/eslint-plugin@3.4.0
├─ @typescript-eslint/parser@3.4.0
├─ eslint@7.3.1
└─ typescript@3.7.4
✨  Done in 2.42s.

> node --version                                                           # service-graph -> feature/RS2-1064-initiate-pickup-with-pin $ ! ?
v12.16.3
```

**Comment by bradzacher** [comment]
This is not a variable - it's an argument.

https://eslint.org/docs/rules/no-unused-vars#argsignorepattern

**Comment by xenoterracide** [comment]
then why does it mark it as such? I included the failure, or is there some other option for arguments to ignore it?

**Comment by bradzacher** [comment]
The linked docs explain the options and how to configure the rule for arguments.

**Comment by xenoterracide** [comment]
ooh, sorry, I missed that when reading in my email... oddly as I made the class bigger it stopped seeing the argument failure anyways, not sure if I can come up with a minimal repro for it though.
